### PR TITLE
Fix #672: output parameters in queries

### LINKED ISSUE #672: Calling SQLServer stored functions (not procs)
Hey! I was wondering if it is possible to call functions and get the return value using this driver?

According to https://docs.microsoft.com/en-us/sql/relational-databases/user-defined-functions/execute-user-defined-functions?view=sql-server-ver15, you must assign the result of the function to a variable. I have tried  many different things, however nothing has worked out for me so far:
```go
	const sqlstr = `@p1 = EXEC test_schema.my_func_with_0_params `
	var res int
	if _, err := db.ExecContext(ctx, sqlstr, sql.Out{Dest: &res}); err != nil {
		return 0, logerror(err)
	}
	return res, nil
}
```

### PR DESCRIPTION
**Pull request by tc-hib**
In RPC, declare output parameters with the "output" keyword.
This is necessary to support output parameters in queries.

### CODE REVIEW

File: mssql.go [hunk 1]
```diff
@@ -577,6 +577,9 @@ func (s *Stmt) makeRPCParams(args []namedValue, isProc bool) ([]param, []string,
 		}
 		params[i+offset].Name = name
 		decls[i] = fmt.Sprintf("%s %s", name, makeDecl(params[i+offset].ti))
```
Comment (kardianos) [inline mssql.go:579]: I would prefer to compose the delcs in a single Sprintf, rather then adding an optional output. Maybe do something like this:
```
const outputSuffix = " output"
var output string
if isOutputValue(val.Value) {
    output = outputSuffix
}
decls[i] = fmt.Sprintf("%s %s%s", name, makeDecl(params[i+offset].ti), output)

```

**Review by kardianos — CHANGES_REQUESTED** [review]
LGTM with change

File: mssql.go (hunk 1 above)
Comment (tc-hib) [reply mssql.go:579]: Where would you declare the const? At the top of mssql.go?

Comment (kardianos) [reply mssql.go:579]: just inline in the func itself. Go will treat it the same anywhere you declare it.

**Review by kardianos — APPROVED** [review]

**Pull request merged**